src/main/java/address/browser/page/GithubProfilePage.java

```diff
@@ -31,11 +35,31 @@ public boolean isValidGithubProfilePage(){
     }
 
     /**
+     * Setup page automation.
+     * Automation tasks: 1) Clicking on the Repositories tab(if not clicked).
+     *                   2) Scrolling to the end of the page when a page is loaded.
+     */
+    public void setupPageAutomation() {
+        if (!wasAutoScrollingSetup) {
+            this.setPageLoadFinishListener(e -> Platform.runLater(this::executePageLoadedTasks));
+            this.setPageAttachedToSceneListener(() -> Platform.runLater(this::executePageLoadedTasks));
+            this.wasAutoScrollingSetup = true;
+            //If page has already been loaded at the point of setting up page automation.
+            //Trigger initial automation.
+            if (!page.isPageLoading()) {
+                this.executePageLoadedTasks();
+            }
+        }
+    }
```

Comment:
I think we can introduce the lock when we need it. Currently we won't have any concurrency issues now